pkg/3scale/amp/operator/base_apimanager_logic_reconciler.go

```diff
@@ -1,42 +1,43 @@
 package operator
 
 import (
-	"context"
-	"fmt"
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/api/policy/v1beta1"
 
 	appsv1alpha1 "github.com/3scale/3scale-operator/pkg/apis/apps/v1alpha1"
 	"github.com/3scale/3scale-operator/pkg/common"
+	"github.com/3scale/3scale-operator/pkg/helper"
+	"github.com/3scale/3scale-operator/pkg/reconcilers"
+
+	"github.com/go-logr/logr"
+	appsv1 "github.com/openshift/api/apps/v1"
+	imagev1 "github.com/openshift/api/image/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
-	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
 type BaseAPIManagerLogicReconciler struct {
-	BaseLogicReconciler
+	*reconcilers.BaseReconciler
 	apiManager *appsv1alpha1.APIManager
+	logger     logr.Logger
```

Comment:
It seems we use this logger now instead of the one contained in reconcilers.BaseReconciler, although we need to create one for reconcilers.BaseReconciler.

What's the idea of having two loggers? Are we assuming the BaseReconciler will be shared among N different kinds of "logic reconcilers" ?

Reply:
The idea is that the logger at `BaseAPIManagerLogicReconciler` contains named value `APIManager Controller=apiManager.Name`. Because `BaseAPIManagerLogicReconciler` life cycle is just one reconciliation loop and has information about APImanager CR. 

On the other hand, BaseReconciler instance life cycle spans to the life of the operator's process.  And its logger is generic about `apimanager.controller` controller.  

[Logr](https://godoc.org/github.com/go-logr/logr) interface allows creating new logger objects out of other more generic logger objects calling `WithValues` or `WithName` methods. And this is used in BaseAPIManagerLogicReconciler constructor:
```
func NewBaseAPIManagerLogicReconciler(b *reconcilers.BaseReconciler, apiManager *appsv1alpha1.APIManager) *BaseAPIManagerLogicReconciler {
    return &BaseAPIManagerLogicReconciler{                                                          
        BaseReconciler: b,                                                                          
        apiManager:     apiManager,                                                                 
        logger:         b.Logger().WithValues("APIManager Controller", apiManager.Name),            
    }                                                                                               
}                                                                                                   
```

Reply:
I see. Makes sense :+1: